app/src/main/java/com/jerboa/ui/components/common/InitializeRoute.kt

```diff
@@ -0,0 +1,26 @@
+package com.jerboa.ui.components.common
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import kotlinx.coroutines.CoroutineScope
+
+@Composable
+fun InitializeRoute(initializationBlock: suspend CoroutineScope.() -> Unit) {
```

Comment:
I have concerns about the `initializeRoute` function. If a Process Death happens, `initializationBlock` won't be re-called because you used `rememberSaveable` for the `isInitialized: Boolean` but the `ViewModel` will be recreated. It seems that could place the `ViewModel` in an inconsistent state.

Reply:
Did not have process death in mind while implementing this. Looks like the isIntialized variable is better stored in the viewModel itself since that is what is currently focusing on here.

But again the dependencies will not be stored which is required for initializing the route. In fact any data in the view model is vulnerable unless we store it in the the saveStateHandle or database ig. However, callbacks cannot be stored anywhere.

One approach we could take is to check if the dependencies exist in the dependency container and redirect to home otherwise. We could prevent crashes due to inconsitent state this way and the UX would also be good. The data in acitivities like CreatePostActivity can be persisted and reloaded when the user navigates back to this activity from the home activity.

What do u think?